internal/cluster/cmd/clusterd/contract.go

```diff
@@ -0,0 +1,482 @@
+		fces[1].V2FileContract = *diff.Revision
+	}
+	cie := applied[len(applied)-1].ChainIndexElement()
+
+	proveTxn := types.V2Transaction{
+		FileContractResolutions: []types.V2FileContractResolution{
+			{
+				Parent:     fces[0],
+				Resolution: new(types.V2FileContractExpiration),
+			},
+			{
+				Parent: fces[1],
+				Resolution: &types.V2StorageProof{
+					ProofIndex: cie,
+					Leaf:       [64]byte{1},
+					Proof:      []types.Hash256{cm.TipState().StorageProofLeafHash([]byte{1})},
+				},
+			},
+		},
+	}
+
+	if _, err := cm.AddV2PoolTransactions(basis, []types.V2Transaction{proveTxn}); err != nil {
+		log.Panic("Failed to add proof and expiration txn to pool", zap.Error(err))
+	}
+
+	tip = cm.Tip()
+	mineBlocks(log, cm, 1)
+
+	if err := syncWallet(cm, w, ws); err != nil {
+		log.Panic("Failed to scan wallet", zap.Error(err))
+	}
+
+	_, applied, err = cm.UpdatesSince(tip, 1000)
+	if err != nil {
+		log.Panic("Failed to retrieve chain updates", zap.Error(err))
+	}
+	for _, cau := range applied {
+		for i := range fces {
+			cau.UpdateElementProof(&fces[i].StateElement)
+		}
+	}
+
+	fc3FinalRev := fc3
+	fc3FinalRev.RevisionNumber = types.MaxRevisionNumber
+	fc3NewContract := fc3
+	fc3NewContract.RevisionNumber++
+	renewal := &types.V2FileContractRenewal{
+		NewContract:       fc3NewContract,
+		FinalRenterOutput: fc3FinalRev.RenterOutput,
+		FinalHostOutput:   fc3FinalRev.HostOutput,
+		RenterRollover:    types.ZeroCurrency,
+		HostRollover:      types.ZeroCurrency,
+	}
+	{
+		cs := cm.TipState()
+		renewal.RenterSignature = renterPrivateKey.SignHash(cs.RenewalSigHash(*renewal))
+		renewal.HostSignature = hostPrivateKey.SignHash(cs.RenewalSigHash(*renewal))
+		renewal.NewContract.RenterSignature = renterPrivateKey.SignHash(cs.ContractSigHash(renewal.NewContract))
+		renewal.NewContract.HostSignature = hostPrivateKey.SignHash(cs.ContractSigHash(renewal.NewContract))
+	}
+	renewalTxn := types.V2Transaction{
+		FileContractResolutions: []types.V2FileContractResolution{{
+			Parent:     fces[2],
+			Resolution: renewal,
+		}},
+	}
+	basis, toSign, err = w.FundV2Transaction(&renewalTxn, fcOut, false)
+	if err != nil {
+		log.Panic("Failed to fund renewal txn", zap.Error(err))
+	}
+	w.SignV2Inputs(&renewalTxn, toSign)
+
+	if _, err := cm.AddV2PoolTransactions(basis, []types.V2Transaction{renewalTxn}); err != nil {
+		log.Panic("Failed to add renewal txn to pool", zap.Error(err))
+	}
+
+	tip = cm.Tip()
+	mineBlocks(log, cm, 1)
+
+	if err := syncWallet(cm, w, ws); err != nil {
+		log.Panic("Failed to scan wallet", zap.Error(err))
+	}
+
+	_, applied, err = cm.UpdatesSince(tip, 1000)
+	if err != nil {
+		log.Panic("Failed to retrieve chain updates", zap.Error(err))
+	}
+	for _, cau := range applied {
+		for i := range fces {
+			cau.UpdateElementProof(&fces[i].StateElement)
+		}
+	}
+
+	id := txn.ID()
+	fc1ID, fc2ID, fc3ID := txn.V2FileContractID(id, 0), txn.V2FileContractID(id, 1), txn.V2FileContractID(id, 2)
+	return FixtureContractAddresses{
```

Comment:
Maybe have these APIs return the whole contracts since we will be fetching them anyway. And just double checking, those contract struct would have ID on them right?

Reply:
Yeah they would